### PR TITLE
tests: add explicit coverage for retry members

### PR DESCRIPTION
**Pull request by tseaver**
See also: #283 

This PR adds explicit coverage for each member of `google.cloud.storage.retry`.  Together with PR #284, should return this package to 100% unit test coverage.

### CODE REVIEW

**Comment by tseaver** [comment]
Flaky unit test reported in https://github.com/googleapis/python-storage/issues/309

**Comment by tseaver** [comment]
Blocked on PR #310.

**Comment by tseaver** [comment]
Rebuilding after merge of #310 (to fix flaky unit test).

**Review by tswast — APPROVED** [review]

**Pull request merged**
